paasta_tools/utils.py

```diff
@@ -2865,13 +2865,17 @@ def build_docker_image_name(service: str) -> str:
     return name
 
 
-def build_docker_tag(service: str, upstream_git_commit: str) -> str:
+def build_docker_tag(
+    service: str, upstream_git_commit: str, image_version: str = None
+) -> str:
     """Builds the DOCKER_TAG string
 
     upstream_git_commit is the SHA that we're building. Usually this is the
     tip of origin/master.
     """
     tag = "{}:paasta-{}".format(build_docker_image_name(service), upstream_git_commit)
```

Comment:
Interesting; since you updated `build_docker_image_version` to optionally accept  `image_version`, why duplicate manually adding `-{image_version}` here?

Reply:
That's a different `build_docker_image_name` that I haven't changed, confusingly there's two of them:
- [The one I changed](https://sourcegraph.yelpcorp.com/Yelp/paasta/-/blob/paasta_tools/generate_deployments_for_service.py?subtree=true#L205)
- [The other one](https://sourcegraph.yelpcorp.com/Yelp/paasta/-/blob/paasta_tools/utils.py?subtree=true#L2855)

Annoyingly, they both do slightly different things 😢

Reply:
That is annoying, you're right! Thanks for clarifying.